provider/src/test/kotlin/org/gradle/kotlin/dsl/resolver/ProjectRootOfTest.kt

```diff
@@ -6,17 +6,27 @@ import org.hamcrest.CoreMatchers.equalTo
 import org.hamcrest.MatcherAssert.assertThat
 
 import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
 
-class ProjectRootOfTest : FolderBasedTest() {
+@RunWith(Parameterized::class)
+class ProjectRootOfTest(private val settingsFileName: String) : FolderBasedTest() {
+
+    companion object {
+        @Parameterized.Parameters(name = "{0}")
+        @JvmStatic
+        fun testCases(): Iterable<Array<Any>> =
```

Comment:
:o: The return type is unnecessary here but if you must declare it, why not `Iterable<Array<String>>`?

Reply:
Because of Junit 4 weirdness, I don't think that will work. I've dealt with Junit 4's Parameterized tests and this can be devilishly tricky to get right.

Reply:
Done.
It does work without the explicit return type, see https://github.com/gradle/kotlin-dsl/pull/541/files#diff-c27c161d1436684b5a027b2b908101e0R18